### ISSUE TITLE
Full-text search in ES|QL

### ISSUE THREAD

**Issue by ChrisHegarty** [elastic/elasticsearch#110131]
Meta issue listing tasks and subtasks for consideration as part of the Initial Search in ES|QL initiative.
Initially the features will be available only on snapshot builds.

- [x] Add support for `MATCH` operator: ` FROM employees | WHERE title MATCH "test"`: @ioanatia  https://github.com/elastic/elasticsearch/pull/110971
   - equivalent to a `MATCH` query
   - add tests to map failures to push to Lucene - at this stage we do not aim for graceful failures, but only to map all the failure cases through tests
- [x] Add support for `MATCH` command: `FROM employees | MATCH "query string"`: @carlosdelest 
  - Initial work:  https://github.com/elastic/elasticsearch/pull/111147
  - Follow up work: https://github.com/elastic/elasticsearch/issues/111487
  - initially this will just default to a `query_string` query
  - add tests to map failures to push to Lucene - the same for `MATCH` operator, we only need to map all failure cases through tests
- [x] Add graceful failures for `MATCH` operator and command:
   - these need to happen before we make any optimizations to the logical plan - we might need a new dedicated place to add these validations, which is why this is a separate issue
- [x] Add dataset for CSV testing for full text search
- [x] Handle `RENAME` with `MATCH` operator https://github.com/elastic/elasticsearch/pull/110971#discussion_r1682591916
- [x] Investigate how to add options to support parameters for operators/commands (check with analytical engine group)
- [x] Add support for scoring: `FROM employees METADATA _score | WHERE title MATCH "test" | SORT _score` @tteofili 
   - adds `_score` as a metadata attribute
   - when `_score` is present it will be altered by queries such as `MATCH`
- [x] Add `qstr` function @carlosdelest #112590 
- [x] Add `match` function
- [x] Remove match command
- [x] #116024 
- [ ] https://github.com/elastic/elasticsearch/issues/115364
- [x] https://github.com/elastic/elasticsearch/issues/115353

Enhacements:

- [ ] support for [query params](https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-rest.html#esql-rest-params) to accept arrays of values (maintains readability for knn)

**Comment by ioanatia** [comment]
With https://github.com/elastic/elasticsearch/pull/110132 merged we should have the basic filtering case covered. I think it was the only piece that was missing. Let me know if there's something else needed, but I am marking that as completed.

**Comment by ioanatia** [comment]
closing this issue - moving the remaining item to the parent meta issue https://github.com/elastic/elasticsearch/issues/123043
